src/ui/public/agg_types/buckets/terms.js

```diff
@@ -103,7 +109,11 @@ export default function TermsAggDefinition(Private) {
             if (!orderBy && prevOrderBy === INIT) {
               // abort until we get the responseValueAggs
               if (!$scope.responseValueAggs) return;
-              params.orderBy = (_.first($scope.responseValueAggs) || { id: 'custom' }).id;
+              let respAgg = _($scope.responseValueAggs).filter((agg) => !$scope.rejectAgg(agg)).first();
+              if (!respAgg) {
+                respAgg = { id: 'custom' };
```

Comment:
How about defaulting to sorting by Term if the existing sort selection is invalid?